build.gradle

```diff
@@ -343,42 +332,27 @@ task setupEclipse {
     group "Setup"
     dependsOn eclipseProject, createEclipseLaunches, resetEclipseDeps
 }
+*/
 
 // RUN SERVER TASKS
 
 compileJava.options.encoding = "UTF-8"
 compileTestJava.options.encoding = "UTF-8"
 
 appengine {
-    httpPort = 8888
-    downloadSdk = true
-    jvmFlags = ["-Xss2m", "-Dfile.encoding=UTF-8",
-                // absolute paths are not supported, the following is relative to the exploded war directory
-                // this only specifies the datastore path, but the indexes are still generated in
-                // WEB-INF/appengine-generated.
-                "-Ddatastore.backing_store=../../local_db.bin"]
```

Comment:
This flag isn't IntelliJ specific; it affects CLI too. Before it was added, the data would get overwritten upon every clean and build. Seems that the path is now relative to the working directory instead of `build/exploded-teammates` though. `"-Ddatastore.backing_store=local_db.bin"` works.

Reply:
Haha here's the full context just for reference: https://github.com/TEAMMATES/teammates/pull/7354/commits/c871380887d591bf1269c38252213a9e48df41a8

Reply:
I've been spending some time trying to make this work for both CLI/Gradle and Eclipse, to no success until now/few hours ago. I should step back a little and make the routine work for CLI/Gradle first, so will add this option back in.

Reply:
@whipermr5 I've done some more experiments, and I feel we may have to discourage `clean` altogether, although I'm quite conflicted on this myself.
Reason: `-Ddatastore.backing_store` keeps data alright, but it won't help saving the uploaded pictures and search indices once they are `clean`ed. It's quite straightforward to make a script to restore search indices, but there's no way to recover the lost pictures. Then again, at the current state pictures have very limited use and one can simply re-upload to make things work again.
What do you think?

Reply:
Apparently there's a `blobstore.backing_store` property: https://stackoverflow.com/a/4811227

Reply:
Ohh the `storagePath` gradle task property looks even more interesting ([link](https://cloud.google.com/appengine/docs/standard/java/tools/gradle-reference)):
> Path to the data (datastore, blobstore, etc.) associated with the application.

Maybe that keeps the indices too?

Reply:
You're looking at version `2-alpha` of the dev server. I believe there's a reason it's being called "alpha": here doing a simple task like adding new instructors can return 3 different exceptions, and I don't really want to invest the time to investigate something that is very uncertain.

Variant 1:
![screen shot 2017-10-01 at 12 29 42 am](https://user-images.githubusercontent.com/7261051/31047508-a90757b6-a63e-11e7-8992-f276f29f1c48.png)

Variant 2:
![screen shot 2017-10-01 at 12 28 02 am](https://user-images.githubusercontent.com/7261051/31047509-a90b78be-a63e-11e7-8e97-ad53fd69ec8d.png)

Variant 3:
![screen shot 2017-10-01 at 12 27 29 am](https://user-images.githubusercontent.com/7261051/31047510-a90bfbfe-a63e-11e7-827b-51c629916e9d.png)